diff --git a/api/@internal/component/ets/action_sheet.d.ts b/api/@internal/component/ets/action_sheet.d.ts
index 3f489ff9f..d12a5da03 100644
--- a/api/@internal/component/ets/action_sheet.d.ts
+++ b/api/@internal/component/ets/action_sheet.d.ts
@@ -20,6 +20,7 @@
 
 /*** if arkts 1.2 */
 import { Resource } from "../../global/resource";
+import { DismissDialogAction } from "./alert_dialog"
 import {
   VoidCallback, ResourceStr, ResourceColor, Dimension, BorderRadiuses, LocalizedEdgeColors, EdgeColors,
   EdgeWidths, LocalizedEdgeWidths, EdgeStyles, LocalizedBorderRadiuses
@@ -148,42 +149,6 @@ interface SheetInfo {
   action: VoidCallback;
 }
 
-/**
- * Component dialog dismiss action.
- *
- * @interface DismissDialogAction
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since arkts {'1.1':'12','1.2':'20'}
- * @arkts 1.1&1.2
- */
-declare interface DismissDialogAction {
-  /**
-   * Defines dialog dismiss function.
-   *
-   * @type { Callback<void> }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'12','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  dismiss: Callback<void>;
-  
-  /**
-   * Dismiss reason type.
-   *
-   * @type { DismissReason }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'12','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  reason: DismissReason;
-}
-
 /**
  * Base button params used for ActionSheet.
  *
diff --git a/api/@internal/component/ets/alert_dialog.d.ts b/api/@internal/component/ets/alert_dialog.d.ts
index 72589404d..094fb613b 100644
--- a/api/@internal/component/ets/alert_dialog.d.ts
+++ b/api/@internal/component/ets/alert_dialog.d.ts
@@ -591,7 +591,7 @@ declare interface AlertDialogButtonOptions extends AlertDialogButtonBaseOptions
  * @since arkts {'1.1':'12','1.2':'20'}
  * @arkts 1.1&1.2
  */
-declare interface TextStyle {
+declare interface TextStyle_alert_dialog {
   /**
    * Set the word break type.
    *
@@ -1106,7 +1106,7 @@ declare interface AlertDialogParam {
    * @since arkts {'1.1':'12','1.2':'20'}
    * @arkts 1.1&1.2
    */
-  textStyle?: TextStyle;
+  textStyle?: TextStyle_alert_dialog;
 
   /**
    * Defines whether to respond to the hover mode.
diff --git a/api/@internal/component/ets/canvas.d.ts b/api/@internal/component/ets/canvas.d.ts
index 3e6779af0..f3fbd726a 100644
--- a/api/@internal/component/ets/canvas.d.ts
+++ b/api/@internal/component/ets/canvas.d.ts
@@ -25,6 +25,7 @@ import { FrameNode } from '../FrameNode';
 import { ImageAIOptions, ImageAnalyzerConfig } from './imageCommon';
 import { Matrix2D } from './matrix2d';
 import { VoidCallback, LengthMetricsUnit, LengthMetrics } from './units';
+import { Size } from '../Graphics'
 /*** endif */
 
 /**
@@ -6145,42 +6146,6 @@ declare class OffscreenCanvas {
   constructor(width: number, height: number, unit: LengthMetricsUnit);
 }
 
-/**
- * Size info.
- *
- * @interface Size
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since arkts {'1.1':'12','1.2':'20'}
- * @arkts 1.1&1.2
- */
-declare interface Size {
-  /**
-   * Defines the width property.
-   *
-   * @type { number }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'12','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  width: number;
-
-  /**
-   * Defines the height property.
-   *
-   * @type { number }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'12','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  height: number;
-}
-
 /**
  * Defines DrawingRenderingContext.
  *
diff --git a/api/@internal/component/ets/common.d.ts b/api/@internal/component/ets/common.d.ts
index b9839fb6e..d09515129 100644
--- a/api/@internal/component/ets/common.d.ts
+++ b/api/@internal/component/ets/common.d.ts
@@ -11465,7 +11465,7 @@ declare interface BorderImageOption {
    * @since arkts {'1.1':'11','1.2':'20'}
    * @arkts 1.1&1.2
    */
-  source?: string | Resource | LinearGradient,
+  source?: string | Resource | LinearGradient_common,
 
   /**
    * Border image width
@@ -16394,7 +16394,8 @@ declare interface StateStyles {
  * @since 20
  * @arkts 1.2
  */
-declare type CustomStyles = (instance: CommonMethod) => void;
+// We can not handle components as arguments in a general way, each case is special
+declare type CustomStyles = (instance: string) => void;
 
 /**
  * Component State Styles.
@@ -28642,7 +28643,7 @@ declare class CommonShapeMethod<T> extends CommonMethod<T> {
  * @since arkts {'1.1':'11','1.2':'20'}
  * @arkts 1.1&1.2
  */
-declare interface LinearGradient {
+declare interface LinearGradient_common {
   /**
    * Linear Gradient Angle
    *
diff --git a/api/@internal/component/ets/custom_dialog_controller.d.ts b/api/@internal/component/ets/custom_dialog_controller.d.ts
index 94f24266a..a66c6d2d7 100644
--- a/api/@internal/component/ets/custom_dialog_controller.d.ts
+++ b/api/@internal/component/ets/custom_dialog_controller.d.ts
@@ -28,6 +28,7 @@ import { Offset, ResourceColor, Dimension, BorderRadiuses, EdgeWidths, EdgeColor
 import { BorderStyle } from './enums'
 import { LengthMetrics } from '../Graphics';
 import { LevelMode,LevelOrder,ImmersiveMode } from '../../@ohos.promptAction'
+import { DismissDialogAction } from './alert_dialog'
 /*** endif */
 
 /**
@@ -711,31 +712,6 @@ declare interface CustomDialogControllerOptions {
  * @since arkts {'1.1':'12','1.2':'20'}
  * @arkts 1.1&1.2
  */
-declare interface DismissDialogAction {
-  /**
-   * Defines dialog dismiss function
-   *
-   * @type { Callback<void> }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'12','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  dismiss: Callback<void>;
-
-  /**
-   * Dismiss reason type.
-   *
-   * @type { DismissReason }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'12','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  reason: DismissReason;
-}
 
 /**
  * Use the CustomDialogController class to display the custom pop-up window.
diff --git a/api/@internal/component/ets/repeat.d.ts b/api/@internal/component/ets/repeat.d.ts
index 6af657932..81c455d53 100644
--- a/api/@internal/component/ets/repeat.d.ts
+++ b/api/@internal/component/ets/repeat.d.ts
@@ -12,275 +12,3 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-/**
- * @file Defines Repeat component.
- * @kit ArkUI
- */
-
-/**
- * Construct a new type for each item.
- *
- * @interface RepeatItem
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 12
- */
-interface RepeatItem<T> {
-  /**
-   * The origin data.
-   *
-   * @type { T }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @form
-   * @atomicservice
-   * @since 12
-   */
-  item: T,
-  /**
-   * index of each item.
-   *
-   * @type { number }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @form
-   * @atomicservice
-   * @since 12
-   */
-  index: number
-}
-
-/**
- * Define the options of repeat virtualScroll to implement reuse and lazy loading.
- *
- * @interface VirtualScrollOptions
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since 12
- */
-interface VirtualScrollOptions {
-  /**
-   * Total data count.
-   *
-   * @type { ?number }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 12
-   */
-  totalCount?: number;
-
-  /**
-   * Reuse or not.
-   *
-   * @type { ?boolean }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 18
-   */
-  reusable?: boolean;
-
-  /**
-   * Data lazy loading
-   *
-   * @param { number } index
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 18
-   */
-  onLazyLoading?(index: number): void;
-
-  /**
-   * The function of total data count.
-   *
-   * @returns { number } Returns the total data count.
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 18
-   */
-  onTotalCount?(): number;
-}
-
-/**
- * Define a builder template option parameter.
- *
- * @interface TemplateOptions
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since 12
- */
-interface TemplateOptions {
-  /**
-   * The cached number of each template.
-   *
-   * @type { ?number }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 12
-   */
-  cachedCount?: number
-}
-
-/**
- * Function that return typed string to render one template.
- *
- * @typedef {function} TemplateTypedFunc<T>
- * @param { T } item - data item.
- * @param {number} index - data index number in array.
- * @returns { string } template type.
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since 12
- */
-declare type TemplateTypedFunc<T> = (item : T, index : number) => string;
-
-/**
- * Define builder function to render one template type.
- *
- * @typedef {function} RepeatItemBuilder<T>
- * @param { RepeatItem<T> } repeatItem - the repeat item builder function.
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since 12
- */
-declare type RepeatItemBuilder<T> = (repeatItem: RepeatItem<T>) => void;
-
-/**
- * Defines the Repeat component attribute functions.
- *
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 12
- */
-/**
- * Defines the Repeat component attribute functions.
- *
- * @extends DynamicNode<RepeatAttribute<T>>
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 18
- */
-declare class RepeatAttribute<T> extends DynamicNode<RepeatAttribute<T>> {
-  /**
-   * Executes itemGenerator of each item.
-   *
-   * @param { function } itemGenerator
-   * @returns { RepeatAttribute<T> }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @form
-   * @atomicservice
-   * @since 12
-   */
-  each(itemGenerator: (repeatItem: RepeatItem<T>) => void): RepeatAttribute<T>;
-  /**
-   * Obtains key of each item.
-   *
-   * @param { function } keyGenerator
-   * @returns { RepeatAttribute<T> }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @form
-   * @atomicservice
-   * @since 12
-   */
-  key(keyGenerator: (item: T, index: number) => string): RepeatAttribute<T>;
-  /**
-   * Enable UI lazy loading when scroll up or down.
-   *
-   * @param { VirtualScrollOptions } virtualScrollOptions that defines the options of repeat virtual scroll to implement reuse and lazy loading.
-   * @returns { RepeatAttribute<T> }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 12
-   */
-  virtualScroll(virtualScrollOptions?: VirtualScrollOptions): RepeatAttribute<T>;
-  /**
-   * Type builder function to render specific type of data item.
-   *
-   * @param { string } type that defines the template id.
-   * @param { RepeatItemBuilder<T> } itemBuilder that defines UI builder function.
-   * @param { TemplateOptions } templateOptions that defines a builder template option parameter.
-   * @returns { RepeatAttribute<T> }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 12
-   */
-  template(type : string, itemBuilder: RepeatItemBuilder<T>, templateOptions?: TemplateOptions): RepeatAttribute<T>;
-  /**
-   * Typed function to render specific type of data item.
-   *
-   * @param { TemplateTypedFunc<T> } typedFunc that define template typed function.
-   * @returns { RepeatAttribute<T> }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 12
-   */
-  templateId(typedFunc: TemplateTypedFunc<T>): RepeatAttribute<T>;
-}
-
-/**
- * Indicates the type of Repeat's Data Source.
- *
- * @typedef { Array<T> | ReadonlyArray<T> | Readonly<Array<T>> }
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 18
- */
-declare type RepeatArray<T> = Array<T> | ReadonlyArray<T> | Readonly<Array<T>>;
-
-/**
- * Indicates the type of Repeat.
- *
- * @typedef { function } RepeatInterface
- * @param { RepeatArray<T> } arr - The Data Source
- * @returns { RepeatAttribute<T> }
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 18
- */
-declare type RepeatInterface = <T>(arr: RepeatArray<T>) => RepeatAttribute<T>;
-
-/**
- * Defines Repeat Component.
- *
- * @type { <T>(arr: Array<T>) => RepeatAttribute<T> }
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 12
- */
-/**
- * Defines Repeat Component, and Add More Array Type.
- *
- * @type { RepeatInterface }
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 18
- */
-declare const Repeat: RepeatInterface;
diff --git a/api/@internal/component/ets/tabs.d.ts b/api/@internal/component/ets/tabs.d.ts
index ddfa74555..da632873b 100644
--- a/api/@internal/component/ets/tabs.d.ts
+++ b/api/@internal/component/ets/tabs.d.ts
@@ -23,6 +23,7 @@ import { BackgroundBlurStyleOptions, BackgroundEffectOptions, BlurStyle, Callbac
 import { EdgeEffect, PageFlipMode } from './enums';
 import { Dimension, Length, ResourceColor } from './units';
 import { CommonModifier } from "../CommonModifier";
+import { DividerStyle } from "./sidebar"
 /*** endif */
 
 /**
@@ -649,114 +650,6 @@ interface TabsInterface {
   (options?: TabsOptions): TabsAttribute;
 }
 
-/**
- * Provides an interface for the style of an divider including stroke width, color, start margin
- * and end margin
- *
- * @interface DividerStyle
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @since 10
- */
-/**
- * Provides an interface for the style of an divider including stroke width, color, start margin
- * and end margin
- *
- * @interface DividerStyle
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since arkts {'1.1':'11','1.2':'20'}
- * @arkts 1.1&1.2
- */
-interface DividerStyle {
-  /**
-   * Define the stroke width of the divider
-   *
-   * @type { Length }
-   * @default 0
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @since 10
-   */
-  /**
-   * Define the stroke width of the divider
-   *
-   * @type { Length }
-   * @default 0
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'11','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  strokeWidth: Length;
-
-  /**
-   * Define the color of the divider
-   *
-   * @type { ?ResourceColor }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @since 10
-   */
-  /**
-   * Define the color of the divider
-   *
-   * @type { ?ResourceColor }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'11','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  color?: ResourceColor;
-
-  /**
-   * Define the start margin of the divider
-   *
-   * @type { ?Length }
-   * @default 0
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @since 10
-   */
-  /**
-   * Define the start margin of the divider
-   *
-   * @type { ?Length }
-   * @default 0
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'11','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  startMargin?: Length;
-
-  /**
-   * Define the end margin of the divider
-   *
-   * @type { ?Length }
-   * @default 0
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @since 10
-   */
-  /**
-   * Define the end margin of the divider
-   *
-   * @type { ?Length }
-   * @default 0
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since arkts {'1.1':'11','1.2':'20'}
-   * @arkts 1.1&1.2
-   */
-  endMargin?: Length;
-}
-
 /**
  * Provides an interface for tabs animation.
  *
diff --git a/api/@internal/component/ets/ui_extension_component.d.ts b/api/@internal/component/ets/ui_extension_component.d.ts
index eac19b4aa..b30176bd1 100644
--- a/api/@internal/component/ets/ui_extension_component.d.ts
+++ b/api/@internal/component/ets/ui_extension_component.d.ts
@@ -23,6 +23,7 @@ import Want from '../../@ohos.app.ability.Want'
 import { Callback, ErrorCallback,BusinessError } from '../../@ohos.base'
 import { CommonMethod } from './common'
 import { ComponentContent } from '../ComponentContent'
+import { TerminationInfo } from './embedded_component'
 /*** endif */
 
 /**
@@ -182,39 +183,6 @@ declare interface TerminationInfo {
    want?: import('../api/@ohos.app.ability.Want').default;
 }
 
-/**
- * Indicates the information when the provider of the embedded UI is terminated.
- *
- * @interface TerminationInfo
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @systemapi
- * @since 20
- * @arkts 1.2
- */
-declare interface TerminationInfo {
-  /**
-   * Defines the termination code.
-   *
-   * @type { number }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @systemapi
-   * @since 20
-   * @arkts 1.2
-   */
-   code: number;
-   
-   /**
-   * Defines the additional termination information.
-   *
-   * @type { ?Want }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @systemapi
-   * @since 20
-   * @arkts 1.2
-   */
-   want?: Want;
-}
-
 /**
  * Get Callback from @ohos.base.
  *
diff --git a/api/@internal/component/ets/xcomponent.d.ts b/api/@internal/component/ets/xcomponent.d.ts
index 06f38f200..227d10a95 100644
--- a/api/@internal/component/ets/xcomponent.d.ts
+++ b/api/@internal/component/ets/xcomponent.d.ts
@@ -229,7 +229,8 @@ declare class XComponentController {
    * @since arkts {'1.1':'12','1.2':'20'}
    * @arkts 1.1&1.2
    */
-  getXComponentSurfaceRotation(): Required<SurfaceRotationOptions>;
+  // type Required does not exists
+  getXComponentSurfaceRotation(): SurfaceRotationOptions;
 
   /**
    * Called after the surface is first created.
diff --git a/api/arkui/ButtonModifier.d.ts b/api/arkui/ButtonModifier.d.ts
index d91f7ea8d..a25c4c25e 100644
--- a/api/arkui/ButtonModifier.d.ts
+++ b/api/arkui/ButtonModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { ButtonAttribute } from './component/Button'
+import { ButtonAttribute } from './component/button'
 /*** endif */
 
 /*** if arkts 1.2 */
diff --git a/api/arkui/CheckboxGroupModifier.d.ts b/api/arkui/CheckboxGroupModifier.d.ts
index 022516fe7..fe821df1b 100644
--- a/api/arkui/CheckboxGroupModifier.d.ts
+++ b/api/arkui/CheckboxGroupModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { CheckboxGroupAttribute } from './component/CheckboxGroup'
+import { CheckboxGroupAttribute } from './component/checkboxgroup'
 /*** endif */
 
 /*** if arkts 1.2 */
diff --git a/api/arkui/CheckboxModifier.d.ts b/api/arkui/CheckboxModifier.d.ts
index e996c16ba..6d3dc02f8 100644
--- a/api/arkui/CheckboxModifier.d.ts
+++ b/api/arkui/CheckboxModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { CheckboxAttribute } from './component/Checkbox'
+import { CheckboxAttribute } from './component/checkbox'
 /*** endif */
 
 /*** if arkts 1.2 */
diff --git a/api/arkui/GridItemModifier.d.ts b/api/arkui/GridItemModifier.d.ts
index 81679e3bb..8bfd84bc3 100644
--- a/api/arkui/GridItemModifier.d.ts
+++ b/api/arkui/GridItemModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { GridItemAttribute } from './component/griditem'
+import { GridItemAttribute } from './component/gridItem'
 /*** endif */
 
 /**
diff --git a/api/arkui/ListItemGroupModifier.d.ts b/api/arkui/ListItemGroupModifier.d.ts
index 1924c8832..a4254c892 100644
--- a/api/arkui/ListItemGroupModifier.d.ts
+++ b/api/arkui/ListItemGroupModifier.d.ts
@@ -19,7 +19,7 @@
  */
 
 /*** if arkts 1.2 */
-import { ListItemGroupAttribute } from './component/ListItemGroup';
+import { ListItemGroupAttribute } from './component/listItemGroup';
 import { AttributeModifier } from './component/common';
 /*** endif */
 
diff --git a/api/arkui/ListItemModifier.d.ts b/api/arkui/ListItemModifier.d.ts
index b165d0d10..8e7fa974a 100644
--- a/api/arkui/ListItemModifier.d.ts
+++ b/api/arkui/ListItemModifier.d.ts
@@ -19,7 +19,7 @@
  */
 
 /*** if arkts 1.2 */
-import { ListItemAttribute } from './component/ListItem';
+import { ListItemAttribute } from './component/listItem';
 import { AttributeModifier } from './component/common';
 /*** endif */
 
diff --git a/api/arkui/ListModifier.d.ts b/api/arkui/ListModifier.d.ts
index 937062f7a..2e19ae449 100644
--- a/api/arkui/ListModifier.d.ts
+++ b/api/arkui/ListModifier.d.ts
@@ -19,8 +19,9 @@
  */
 
 /*** if arkts 1.2 */
-import { ListItemAttribute } from './component/ListItem';
+import { ListItemAttribute } from './component/listItem';
 import { AttributeModifier } from './component/common';
+import { ListAttribute } from './component/list'
 /*** endif */
 
 /**
diff --git a/api/arkui/MenuItemModifier.d.ts b/api/arkui/MenuItemModifier.d.ts
index 2b9b25a23..ed366d1a9 100644
--- a/api/arkui/MenuItemModifier.d.ts
+++ b/api/arkui/MenuItemModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { MenuItemAttribute } from './component/MenuItem'
+import { MenuItemAttribute } from './component/menuItem'
 /*** endif */
 
 /*** if arkts 1.2 */
diff --git a/api/arkui/MenuModifier.d.ts b/api/arkui/MenuModifier.d.ts
index a97574b22..d52490699 100644
--- a/api/arkui/MenuModifier.d.ts
+++ b/api/arkui/MenuModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { MenuAttribute } from './component/Menu'
+import { MenuAttribute } from './component/menu'
 /*** endif */
 
 /*** if arkts 1.2 */
diff --git a/api/arkui/NavRouterModifier.d.ets b/api/arkui/NavRouterModifier.d.ets
index 90ba19ec7..5a6051e37 100644
--- a/api/arkui/NavRouterModifier.d.ets
+++ b/api/arkui/NavRouterModifier.d.ets
@@ -20,7 +20,8 @@
 
 
 import { AttributeModifier } from './component/common'
-import { NavRouterAttribute } from './component/navRouter'
+// no such file
+// import { NavRouterAttribute } from './component/navRouter'
 
 
 
@@ -33,7 +34,7 @@ import { NavRouterAttribute } from './component/navRouter'
  * @atomicservice
  * @since 12
 */
-export declare class NavRouterModifier extends NavRouterAttribute implements AttributeModifier<NavRouterAttribute> {
+export declare class NavRouterModifier implements AttributeModifier<object> {
 
   /**
    * Defines the normal update attribute function.
@@ -44,9 +45,9 @@ export declare class NavRouterModifier extends NavRouterAttribute implements Att
    * @atomicservice
    * @since 12
    */
-  applyNormalAttribute: undefined | ((instance: NavRouterAttribute) => void);
-  applyPressedAttribute: undefined | ((instance: NavRouterAttribute) => void);
-  applyFocusedAttribute: undefined | ((instance: NavRouterAttribute) => void);
-  applyDisabledAttribute: undefined | ((instance: NavRouterAttribute) => void);
-  applySelectedAttribute: undefined | ((instance: NavRouterAttribute) => void);
+  // applyNormalAttribute: undefined | ((instance: NavRouterAttribute) => void);
+  // applyPressedAttribute: undefined | ((instance: NavRouterAttribute) => void);
+  // applyFocusedAttribute: undefined | ((instance: NavRouterAttribute) => void);
+  // applyDisabledAttribute: undefined | ((instance: NavRouterAttribute) => void);
+  // applySelectedAttribute: undefined | ((instance: NavRouterAttribute) => void);
 }
diff --git a/api/arkui/NavigatorModifier.d.ets b/api/arkui/NavigatorModifier.d.ets
index 55d2ea997..57995a69b 100644
--- a/api/arkui/NavigatorModifier.d.ets
+++ b/api/arkui/NavigatorModifier.d.ets
@@ -20,7 +20,8 @@
 
 
 import { AttributeModifier } from './component/common'
-import { NavigatorAttribute } from './component/navigator'
+// file does not exists
+// import { NavigatorAttribute } from './component/navigator'
 
 
 /**
@@ -32,7 +33,7 @@ import { NavigatorAttribute } from './component/navigator'
  * @atomicservice
  * @since 12
 */
-export declare class NavigatorModifier extends NavigatorAttribute implements AttributeModifier<NavigatorAttribute> {
+export declare class NavigatorModifier implements AttributeModifier<object> {
 
   /**
    * Defines the normal update attribute function.
@@ -43,9 +44,9 @@ export declare class NavigatorModifier extends NavigatorAttribute implements Att
    * @atomicservice
    * @since 12
    */
-  applyNormalAttribute: undefined | ((instance: NavigatorAttribute) => void);
-  applyPressedAttribute: undefined | ((instance: NavigatorAttribute) => void);
-  applyFocusedAttribute: undefined | ((instance: NavigatorAttribute) => void);
-  applyDisabledAttribute: undefined | ((instance: NavigatorAttribute) => void);
-  applySelectedAttribute: undefined | ((instance: NavigatorAttribute) => void);
+  // applyNormalAttribute: undefined | ((instance: NavigatorAttribute) => void);
+  // applyPressedAttribute: undefined | ((instance: NavigatorAttribute) => void);
+  // applyFocusedAttribute: undefined | ((instance: NavigatorAttribute) => void);
+  // applyDisabledAttribute: undefined | ((instance: NavigatorAttribute) => void);
+  // applySelectedAttribute: undefined | ((instance: NavigatorAttribute) => void);
 }
diff --git a/api/arkui/PanelModifier.d.ets b/api/arkui/PanelModifier.d.ets
index 75d34cf8f..506205258 100644
--- a/api/arkui/PanelModifier.d.ets
+++ b/api/arkui/PanelModifier.d.ets
@@ -19,7 +19,8 @@
  */
 
 import { AttributeModifier } from './component/common'
-import { PanelAttribute } from './component/panel'
+// no such file
+// import { PanelA ttribute } from './component/panel'
 
 
 
@@ -32,7 +33,7 @@ import { PanelAttribute } from './component/panel'
  * @atomicservice
  * @since 12
 */
-export declare class PanelModifier extends PanelAttribute implements AttributeModifier<PanelAttribute> {
+export declare class PanelModifier implements AttributeModifier<PanelAttribute> {
 
   /**
    * Defines the normal update attribute function.
@@ -43,9 +44,9 @@ export declare class PanelModifier extends PanelAttribute implements AttributeMo
    * @atomicservice
    * @since 12
    */
-  applyNormalAttribute: undefined | ((instance: PanelAttribute) => void);
-  applyPressedAttribute: undefined | ((instance: PanelAttribute) => void);
-  applyFocusedAttribute: undefined | ((instance: PanelAttribute) => void);
-  applyDisabledAttribute: undefined | ((instance: PanelAttribute) => void);
-  applySelectedAttribute: undefined | ((instance: PanelAttribute) => void);
+  // applyNormalAttribute: undefined | ((instance: PanelAttribute) => void);
+  // applyPressedAttribute: undefined | ((instance: PanelAttribute) => void);
+  // applyFocusedAttribute: undefined | ((instance: PanelAttribute) => void);
+  // applyDisabledAttribute: undefined | ((instance: PanelAttribute) => void);
+  // applySelectedAttribute: undefined | ((instance: PanelAttribute) => void);
 }
diff --git a/api/arkui/RadioModifier.d.ts b/api/arkui/RadioModifier.d.ts
index 173274fce..066d2caa1 100644
--- a/api/arkui/RadioModifier.d.ts
+++ b/api/arkui/RadioModifier.d.ts
@@ -20,7 +20,8 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { RadioAttribute } from './component/Radio'
+// does not exists
+// import { RadioAttribute } from './component/Radio'
 /*** endif */
 
 /*** if arkts 1.2 */
@@ -35,7 +36,8 @@ import { RadioAttribute } from './component/Radio'
  * @since 20
  * @arkts 1.2
  */
-declare type RadioApplyNormalFunction = (RadioAttribute: RadioAttribute) => void;
+// does not used
+// declare type RadioApplyNormalFunction = (RadioAttribute: RadioAttribute) => void;
 
 /**
  * Defines Radio Modifier
@@ -47,62 +49,63 @@ declare type RadioApplyNormalFunction = (RadioAttribute: RadioAttribute) => void
  * @since 20
  * @arkts 1.2
 */
-export declare class RadioAttribute extends RadioAttribute implements AttributeModifier<RadioAttribute> {
+// what it was? `class A extends A`?
+export declare class RadioAttributeModifier implements AttributeModifier<object> {
   /**
    * Defines the normal update attribute function.
-   * 
-   * @type {undefined | ((instance: RadioAttribute) => void)}
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 20
-   * @arkts 1.2
-   */
-  applyNormalAttribute: undefined | ((instance: RadioAttribute) => void);
-  /**
-   * Defines the pressed update attribute function.
-   * 
-   * @type {undefined | ((instance: RadioAttribute) => void)}
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 20
-   * @arkts 1.2
-   */
-  applyPressedAttribute: undefined | ((instance: RadioAttribute) => void);
-  /**
-   * Defines the focused update attribute function.
-   * 
-   * @type {undefined | ((instance: RadioAttribute) => void)}
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 20
-   * @arkts 1.2
-   */
-  applyFocusedAttribute: undefined | ((instance: RadioAttribute) => void);
-  /**
-   * Defines the disabled update attribute function.
-   * 
-   * @type {undefined | ((instance: RadioAttribute) => void)}
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 20
-   * @arkts 1.2
-   */
-  applyDisabledAttribute: undefined | ((instance: RadioAttribute) => void);
-  /**
-   * Defines the selected update attribute function.
-   * 
-   * @type {undefined | ((instance: RadioAttribute) => void)}
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 20
-   * @arkts 1.2
-   */
-  applySelectedAttribute: undefined | ((instance: RadioAttribute) => void);
+  //  * 
+  //  * @type {undefined | ((instance: RadioAttribute) => void)}
+  //  * @syscap SystemCapability.ArkUI.ArkUI.Full
+  //  * @crossplatform
+  //  * @atomicservice
+  //  * @since 20
+  //  * @arkts 1.2
+  //  */
+  // applyNormalAttribute: undefined | ((instance: RadioAttribute) => void);
+  // /**
+  //  * Defines the pressed update attribute function.
+  //  * 
+  //  * @type {undefined | ((instance: RadioAttribute) => void)}
+  //  * @syscap SystemCapability.ArkUI.ArkUI.Full
+  //  * @crossplatform
+  //  * @atomicservice
+  //  * @since 20
+  //  * @arkts 1.2
+  //  */
+  // applyPressedAttribute: undefined | ((instance: RadioAttribute) => void);
+  // /**
+  //  * Defines the focused update attribute function.
+  //  * 
+  //  * @type {undefined | ((instance: RadioAttribute) => void)}
+  //  * @syscap SystemCapability.ArkUI.ArkUI.Full
+  //  * @crossplatform
+  //  * @atomicservice
+  //  * @since 20
+  //  * @arkts 1.2
+  //  */
+  // applyFocusedAttribute: undefined | ((instance: RadioAttribute) => void);
+  // /**
+  //  * Defines the disabled update attribute function.
+  //  * 
+  //  * @type {undefined | ((instance: RadioAttribute) => void)}
+  //  * @syscap SystemCapability.ArkUI.ArkUI.Full
+  //  * @crossplatform
+  //  * @atomicservice
+  //  * @since 20
+  //  * @arkts 1.2
+  //  */
+  // applyDisabledAttribute: undefined | ((instance: RadioAttribute) => void);
+  // /**
+  //  * Defines the selected update attribute function.
+  //  * 
+  //  * @type {undefined | ((instance: RadioAttribute) => void)}
+  //  * @syscap SystemCapability.ArkUI.ArkUI.Full
+  //  * @crossplatform
+  //  * @atomicservice
+  //  * @since 20
+  //  * @arkts 1.2
+  //  */
+  // applySelectedAttribute: undefined | ((instance: RadioAttribute) => void);
 }
 /*** endif */
 
@@ -115,7 +118,7 @@ export declare class RadioAttribute extends RadioAttribute implements AttributeM
  * @atomicservice
  * @since 12
 */
-export declare class RadioModifier extends RadioAttribute implements AttributeModifier<RadioAttribute> {
+export declare class RadioModifier implements AttributeModifier<object> {
 
   /**
    * Defines the normal update attribute function.
@@ -126,5 +129,5 @@ export declare class RadioModifier extends RadioAttribute implements AttributeMo
    * @atomicservice
    * @since 12
    */
-  applyNormalAttribute?(instance: RadioAttribute): void;
+  // applyNormalAttribute?(instance: RadioAttribute): void;
 }
\ No newline at end of file
diff --git a/api/arkui/RatingModifier.d.ts b/api/arkui/RatingModifier.d.ts
index acd2d9663..e9d7c9e02 100644
--- a/api/arkui/RatingModifier.d.ts
+++ b/api/arkui/RatingModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { RatingAttribute } from './component/Rating'
+import { RatingAttribute } from './component/rating'
 /*** endif */
 
 /*** if arkts 1.2 */
diff --git a/api/arkui/RichEditorModifier.d.ts b/api/arkui/RichEditorModifier.d.ts
index 30f1e8249..7acc87de7 100644
--- a/api/arkui/RichEditorModifier.d.ts
+++ b/api/arkui/RichEditorModifier.d.ts
@@ -20,7 +20,6 @@
 
 
 
-
 /**
  * Defines RichEditor Modifier
  *
@@ -30,41 +29,42 @@
  * @atomicservice
  * @since 12
 */
-export declare class RichEditorModifier extends RichEditorAttribute implements AttributeModifier<RichEditorAttribute> {
+// no imports, two same declarations
+// export declare class RichEditorModifier extends RichEditorAttribute implements AttributeModifier<RichEditorAttribute> {
 
-  /**
-   * Defines the normal update attribute function.
-   * 
-   * @param { RichEditorAttribute } instance
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 12
-   */
-  applyNormalAttribute?(instance: RichEditorAttribute): void;
-}
+//   /**
+//    * Defines the normal update attribute function.
+//    * 
+//    * @param { RichEditorAttribute } instance
+//    * @syscap SystemCapability.ArkUI.ArkUI.Full
+//    * @crossplatform
+//    * @atomicservice
+//    * @since 12
+//    */
+//   applyNormalAttribute?(instance: RichEditorAttribute): void;
+// }
 
-/**
- * Defines RichEditor Modifier
- *
- * @extends RichEditorAttribute
- * @implements AttributeModifier
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @atomicservice
- * @since 20
- * @arkts 1.2
-*/
-export declare class RichEditorModifier extends RichEditorAttribute implements AttributeModifier<RichEditorAttribute> {
+// /**
+//  * Defines RichEditor Modifier
+//  *
+//  * @extends RichEditorAttribute
+//  * @implements AttributeModifier
+//  * @syscap SystemCapability.ArkUI.ArkUI.Full
+//  * @atomicservice
+//  * @since 20
+//  * @arkts 1.2
+// */
+// export declare class RichEditorModifier extends RichEditorAttribute implements AttributeModifier<RichEditorAttribute> {
 
-  /**
-   * Defines the normal update attribute function.
-   * 
-   * @type { ?((instance: RichEditorAttribute) => void) }
-   * @syscap SystemCapability.ArkUI.ArkUI.Full
-   * @crossplatform
-   * @atomicservice
-   * @since 20
-   * @arkts 1.2
-   */
-  applyNormalAttribute?: ((instance: RichEditorAttribute) => void);
-}
+//   /**
+//    * Defines the normal update attribute function.
+//    * 
+//    * @type { ?((instance: RichEditorAttribute) => void) }
+//    * @syscap SystemCapability.ArkUI.ArkUI.Full
+//    * @crossplatform
+//    * @atomicservice
+//    * @since 20
+//    * @arkts 1.2
+//    */
+//   applyNormalAttribute?: ((instance: RichEditorAttribute) => void);
+// }
diff --git a/api/arkui/SelectModifier.d.ts b/api/arkui/SelectModifier.d.ts
index 42f355237..c2ededdda 100644
--- a/api/arkui/SelectModifier.d.ts
+++ b/api/arkui/SelectModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { SelectAttribute } from './component/Select'
+import { SelectAttribute } from './component/select'
 /*** endif */
 
 /*** if arkts 1.2 */
diff --git a/api/arkui/SliderModifier.d.ts b/api/arkui/SliderModifier.d.ts
index 504521302..69bd9d390 100644
--- a/api/arkui/SliderModifier.d.ts
+++ b/api/arkui/SliderModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { SliderAttribute } from './component/Slider'
+import { SliderAttribute } from './component/slider'
 /*** endif */
 
 /*** if arkts 1.2 */
diff --git a/api/arkui/ToggleModifier.d.ts b/api/arkui/ToggleModifier.d.ts
index 38d642bb9..78ac1d5de 100644
--- a/api/arkui/ToggleModifier.d.ts
+++ b/api/arkui/ToggleModifier.d.ts
@@ -20,7 +20,7 @@
 
 /*** if arkts 1.2 */
 import { AttributeModifier } from './component/common'
-import { SelectAttribute } from './component/Select'
+import { SelectAttribute } from './component/select'
 /*** endif */
 
 /*** if arkts 1.2 */
diff --git a/api/arkui/component/repeat.d.ets b/api/arkui/component/repeat.d.ets
index 58925993a..7e37c67ec 100755
--- a/api/arkui/component/repeat.d.ets
+++ b/api/arkui/component/repeat.d.ets
@@ -17,242 +17,3 @@
  * @kit ArkUI
  * @arkts 1.2
  */
-
-
-import { memo, __memo_context_type, __memo_id_type } from './../stateManagement/runtime'
-
-/**
- * Indicates the type of Repeat's Data Source.
- *
- * @typedef { Array<T> | ReadonlyArray<T> | Readonly<Array<T>> }
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 20
- */
-type RepeatArray<T> = Array<T> | ReadonlyArray<T> | Readonly<Array<T>>;
-/**
- * Function that return typed string to render one template.
- *
- * @typedef {function} TemplateTypedFunc<T>
- * @param { T } item - data item.
- * @param {number} index - data index number in array.
- * @returns { string } template type.
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since 20
- */
-type TemplateTypedFunc<T> = (item: T, index: number) => string;
-/**
- * Define builder function to render one template type.
- *
- * @typedef {function} RepeatItemBuilder<T>
- * @param { RepeatItem<T> } repeatItem - the repeat item builder function.
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since 20
- */
-type RepeatItemBuilder<T> = (repeatItem: RepeatItem<T>) => void;
-
-/**
- * Construct a new type for each item.
- *
- * @interface RepeatItem
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 20
- */
-export interface RepeatItem<T> {
-    /**
-     * The origin data.
-     *
-     * @type { T }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @form
-     * @atomicservice
-     * @since 20
-     */
-    item: T;
-    /**
-     * index of each item.
-     *
-     * @type { number }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @form
-     * @atomicservice
-     * @since 20
-     */
-    index: number;
-}
-
-/**
- * Define the options of repeat virtualScroll to implement reuse and lazy loading.
- *
- * @interface VirtualScrollOptions
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since 20
- */
-export interface VirtualScrollOptions {
-    /**
-     * Total data count.
-     *
-     * @type { ?number }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @atomicservice
-     * @since 20
-     */
-    totalCount?: number;
-    /**
-     * Reuse or not.
-     *
-     * @type { ?boolean }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @atomicservice
-     * @since 20
-     */
-    reusable?: boolean;
-    /**
-     * Data lazy loading
-     *
-     * @param { number } index
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @atomicservice
-     * @since 20
-     */
-    onLazyLoading?: (index: number) => void;
-    /**
-     * The function of total data count.
-     *
-     * @returns { number } Returns the total data count.
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @atomicservice
-     * @since 20
-     */
-    onTotalCount?: () => number;
-}
-
-/**
- * Define a builder template option parameter.
- *
- * @interface TemplateOptions
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @atomicservice
- * @since 20
- */
-export interface TemplateOptions {
-    /**
-     * The cached number of each template.
-     *
-     * @type { ?number }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @atomicservice
-     * @since 20
-     */
-    cachedCount?: number;
-}
-
-/**
- * Defines the Repeat component attribute functions.
- *
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 20
- */
-export interface UIRepeatAttribute<T> {
-    /**
-     * Executes itemGenerator of each item.
-     *
-     * @param { RepeatItemBuilder<T> } itemGenerator
-     * @returns { UIRepeatAttribute<T> }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @form
-     * @atomicservice
-     * @since 20
-     */
-    @memo
-    each(itemGenerator: RepeatItemBuilder<T>): this;
-    /**
-     * Obtains key of each item.
-     *
-     * @param { function } keyGenerator
-     * @returns { UIRepeatAttribute<T> }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @form
-     * @atomicservice
-     * @since 20
-     */
-    @memo
-    key(keyGenerator: (item: T, index: number) => string): this;
-    /**
-     * Enable UI lazy loading when scroll up or down.
-     *
-     * @param { VirtualScrollOptions } virtualScrollOptions that defines the options of repeat virtual scroll to implement reuse and lazy loading.
-     * @returns { UIRepeatAttribute<T> }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @atomicservice
-     * @since 20
-     */
-    @memo
-    virtualScroll(virtualScrollOptions?: VirtualScrollOptions): this;
-    /**
-     * Type builder function to render specific type of data item.
-     *
-     * @param { string } type that defines the template id.
-     * @param { RepeatItemBuilder<T> } itemBuilder that defines UI builder function.
-     * @param { TemplateOptions } templateOptions that defines a builder template option parameter.
-     * @returns { UIRepeatAttribute<T> }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @atomicservice
-     * @since 20
-     */
-    @memo
-    template(type: string, itemBuilder: RepeatItemBuilder<T>, templateOptions?: TemplateOptions): this;
-    /**
-     * Typed function to render specific type of data item.
-     *
-     * @param { TemplateTypedFunc<T> } typedFunc that define template typed function.
-     * @returns { UIRepeatAttribute<T> }
-     * @syscap SystemCapability.ArkUI.ArkUI.Full
-     * @crossplatform
-     * @atomicservice
-     * @since 20
-     */
-    @memo
-    templateId(typedFunc: TemplateTypedFunc<T>): this;
-}
-
-/**
- * Indicates the type of Repeat.
- *
- * @typedef { function } RepeatInterface
- * @param { RepeatArray<T> } arr - The Data Source
- * @returns { UIRepeatAttribute<T> }
- * @syscap SystemCapability.ArkUI.ArkUI.Full
- * @crossplatform
- * @form
- * @atomicservice
- * @since 20
- */
-@memo
-export declare function Repeat<T>(arr: RepeatArray<T>): UIRepeatAttribute<T>
\ No newline at end of file
diff --git a/build-tools/package-lock.json b/build-tools/package-lock.json
index 516ab7e28..c75e47009 100644
--- a/build-tools/package-lock.json
+++ b/build-tools/package-lock.json
@@ -1,51 +1,96 @@
 {
   "name": "delete",
   "version": "1.0.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 3,
   "requires": true,
-  "dependencies": {
-    "fs": {
+  "packages": {
+    "": {
+      "name": "delete",
+      "version": "1.0.0",
+      "hasInstallScript": true,
+      "license": "ISC",
+      "dependencies": {
+        "commander": "^13.1.0",
+        "fs": "^0.0.1-security",
+        "path": "^0.12.7",
+        "typescript": "npm:ohos-typescript@4.9.5-r5"
+      }
+    },
+    "node_modules/commander": {
+      "version": "13.1.0",
+      "resolved": "https://repo.huaweicloud.com/repository/npm/commander/-/commander-13.1.0.tgz",
+      "integrity": "sha512-/rFeCpNJQbhSZjGVwO9RFV3xPqbnERS8MmIQzCtD/zl6gpJuV/bMLuN92oG3F7d8oDEHHRrujSXNUr8fpjntKw==",
+      "license": "MIT",
+      "engines": {
+        "node": ">=18"
+      }
+    },
+    "node_modules/fs": {
       "version": "0.0.1-security",
       "resolved": "https://repo.huaweicloud.com/repository/npm/fs/-/fs-0.0.1-security.tgz",
-      "integrity": "sha512-3XY9e1pP0CVEUCdj5BmfIZxRBTSDycnbqhIOGec9QYtmVH2fbLpj86CFWkrNOkt/Fvty4KZG5lTglL9j/gJ87w=="
+      "integrity": "sha512-3XY9e1pP0CVEUCdj5BmfIZxRBTSDycnbqhIOGec9QYtmVH2fbLpj86CFWkrNOkt/Fvty4KZG5lTglL9j/gJ87w==",
+      "license": "ISC"
     },
-    "inherits": {
+    "node_modules/inherits": {
       "version": "2.0.3",
       "resolved": "https://repo.huaweicloud.com/repository/npm/inherits/-/inherits-2.0.3.tgz",
-      "integrity": "sha512-x00IRNXNy63jwGkJmzPigoySHbaqpNuzKbBOmzK+g2OdZpQ9w+sxCN+VSB3ja7IAge2OP2qpfxTjeNcyjmW1uw=="
+      "integrity": "sha512-x00IRNXNy63jwGkJmzPigoySHbaqpNuzKbBOmzK+g2OdZpQ9w+sxCN+VSB3ja7IAge2OP2qpfxTjeNcyjmW1uw==",
+      "license": "ISC"
     },
-    "json5": {
+    "node_modules/json5": {
       "version": "2.2.3",
       "resolved": "https://repo.huaweicloud.com/repository/npm/json5/-/json5-2.2.3.tgz",
-      "integrity": "sha512-XmOWe7eyHYH14cLdVPoyg+GOH3rYX++KpzrylJwSW98t3Nk+U8XOl8FWKOgwtzdb8lXGf6zYwDUzeHMWfxasyg=="
+      "integrity": "sha512-XmOWe7eyHYH14cLdVPoyg+GOH3rYX++KpzrylJwSW98t3Nk+U8XOl8FWKOgwtzdb8lXGf6zYwDUzeHMWfxasyg==",
+      "license": "MIT",
+      "bin": {
+        "json5": "lib/cli.js"
+      },
+      "engines": {
+        "node": ">=6"
+      }
     },
-    "path": {
+    "node_modules/path": {
       "version": "0.12.7",
       "resolved": "https://repo.huaweicloud.com/repository/npm/path/-/path-0.12.7.tgz",
       "integrity": "sha512-aXXC6s+1w7otVF9UletFkFcDsJeO7lSZBPUQhtb5O0xJe8LtYhj/GxldoL09bBj9+ZmE2hNoHqQSFMN5fikh4Q==",
-      "requires": {
+      "license": "MIT",
+      "dependencies": {
         "process": "^0.11.1",
         "util": "^0.10.3"
       }
     },
-    "process": {
+    "node_modules/process": {
       "version": "0.11.10",
       "resolved": "https://repo.huaweicloud.com/repository/npm/process/-/process-0.11.10.tgz",
-      "integrity": "sha512-cdGef/drWFoydD1JsMzuFf8100nZl+GT+yacc2bEced5f9Rjk4z+WtFUTBu9PhOi9j/jfmBPu0mMEY4wIdAF8A=="
-    },
-    "typescript": {
-      "version": "npm:ohos-typescript@4.2.3-r5",
-      "resolved": "https://repo.huaweicloud.com/repository/npm/ohos-typescript/-/ohos-typescript-4.2.3-r5.tgz",
-      "integrity": "sha512-9l72RIvrz8RMpU8j1uK7cIL5/gpNR11nelIFOmMDDOA3LJf70NlMgVIXOIg7MjCgjPb+U9PDwYBpalvb+rd2LA==",
-      "requires": {
-        "json5": "^2.2.3"
+      "integrity": "sha512-cdGef/drWFoydD1JsMzuFf8100nZl+GT+yacc2bEced5f9Rjk4z+WtFUTBu9PhOi9j/jfmBPu0mMEY4wIdAF8A==",
+      "license": "MIT",
+      "engines": {
+        "node": ">= 0.6.0"
+      }
+    },
+    "node_modules/typescript": {
+      "name": "ohos-typescript",
+      "version": "4.9.5-r5",
+      "resolved": "https://repo.huaweicloud.com/repository/npm/ohos-typescript/-/ohos-typescript-4.9.5-r5.tgz",
+      "integrity": "sha512-jaavgk6VJBs/Dwllj6y97+b4ve3CZPjl7uWAOqTsGM+2uaA6XIxxVgZZJdo811bccKhDkjvuqkUcHfSDidV8gw==",
+      "license": "Apache-2.0",
+      "dependencies": {
+        "json5": "2.2.3"
+      },
+      "bin": {
+        "tsc": "bin/tsc",
+        "tsserver": "bin/tsserver"
+      },
+      "engines": {
+        "node": ">=4.2.0"
       }
     },
-    "util": {
+    "node_modules/util": {
       "version": "0.10.4",
       "resolved": "https://repo.huaweicloud.com/repository/npm/util/-/util-0.10.4.tgz",
       "integrity": "sha512-0Pm9hTQ3se5ll1XihRic3FDIku70C+iHUdT/W926rSgHV5QgXsYbKZN8MSC3tJtSkhuROzvsQjAaFENRXr+19A==",
-      "requires": {
+      "license": "MIT",
+      "dependencies": {
         "inherits": "2.0.3"
       }
     }
